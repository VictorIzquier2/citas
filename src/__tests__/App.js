import React from 'react';
import {render, screen} from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';

test('La aplicación funciona', () => {
  render(<App />);

  expect(screen.getByText('Administrador de pacientes')).toBeInTheDocument();
  expect(screen.getByTestId('nombreApp').textContent).toBe('Administrador de pacientes');
  expect(screen.getByTestId('nombreApp').tagName).toBe('H1');

  expect(screen.getByText('Solicita una cita')).toBeInTheDocument();

});

test('<App /> Agregar una cita y verificar el Heading', () => {

  render(<App />);

  userEvent.type(screen.getByTestId('mascota'), 'Robin')
  userEvent.type(screen.getByTestId('propietario'), 'Victor');
  userEvent.type(screen.getByTestId('fecha'), '2021-09-10');
  userEvent.type(screen.getByTestId('hora'), '10:30');
  userEvent.type(screen.getByTestId('sintomas'), 'Amor');

  // Click en el botón de submit
  const btnSubmit = screen.getByTestId('btn-submit');
  userEvent.click(btnSubmit);
  
  // Revisar que no se produzca ninguna alerta 
  const alerta = screen.queryByTestId('alerta');
  expect(alerta).not.toBeInTheDocument();

  // Revisar el título dinámico
  expect(screen.getByTestId('titulo-dinamico').textContent).toBe('Administra tus citas');

});

test('<App /> Verificar las citas en el DOM', async () => {
  render(<App />);

  const citas = await screen.findAllByTestId('cita');

  // Snapshot crea un archivo para verificar su contenido
  //expect(citas).toMatchSnapshot();

  expect(screen.getByTestId('btn-eliminar').tagName).toBe('BUTTON');

});