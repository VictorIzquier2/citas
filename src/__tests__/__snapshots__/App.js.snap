// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Verificar las citas en el DOM 1`] = `
Array [
  <div
    class="cita"
    data-testid="cita"
  >
    <p>
      Mascota:
      <span>
        Robin
      </span>
    </p>
    <p>
      Propietario: 
      <span>
        Victor
      </span>
    </p>
    <p>
      Fecha: 
      <span>
        2021-09-10
      </span>
    </p>
    <p>
      Hora: 
      <span>
        10:30
      </span>
    </p>
    <p>
      Síntomas: 
      <span>
        Amor
      </span>
    </p>
    <button
      class="button eliminar u-full-width"
    >
      Eliminar ×
    </button>
  </div>,
]
`;
